File: backend/DTO/Car/AddCarDto.cs
using System.ComponentModel.DataAnnotations;
using DreamBid.Validation;

namespace DreamBid.Dtos.Car
{
    public class AddCarDto
    {
        [Required]
        [MaxLength(50)]
        public string Make { get; set; }

        [Required]
        [MaxLength(50)]
        public string Model { get; set; }

        [Required]
        [CarManufacturedYear]
        public int Year { get; set; }

        [Required]
        [Range(0, 1000000)]
        public double Mileage { get; set; }

        [Required]
        [MaxLength(20)]
        public string VIN { get; set; }


        public string ConditionReport { get; set; }

        [Required]
        public double StartingPrice { get; set; }

        public double? ReservePrice { get; set; }
    }
}

File: frontend/src/dto/car/addCarDto.js
import Joi from "joi";

class AddCarDto {

    make = "";
    model = "";
    year = "";
    mileage ="";
    vin = "";
    conditionReport = "";
    startingPrice = "";
    reservePrice = "";

}


export function validateAddCarDto(addCarDto){

    const schema = Joi.object({
   
        make: Joi.string().max(50).required(),
        model: Joi.string().max(50).required(),
        year: Joi.number().integer().min(1886).max(new Date().getFullYear()),
        mileage: Joi.number().min(0).max(1000000).required(),
        vin: Joi.string().max(20).required(),
        conditionReport: Joi.string(),
        startingPrice: Joi.number().required(),
        reservePrice: Joi.number(),

    })
    return schema.validate(addCarDto);
}

export default AddCarDto;